pkg/build/controller/build/build_controller.go

```diff
@@ -1233,11 +1237,30 @@ func (bc *BuildController) createBuildPod(build *buildv1.Build) (*buildUpdate, e
 		if err != nil {
 			return nil, err
 		}
+		// HashOwnerReference does not distinguish between having no refs to compare with the build and
+		// having a different ref compared to our build here;  that said, if the pod has not owner refs,
+		// there is no chance GC will clean it up, so bypass our retry here
 		if !strategy.HasOwnerReference(existingPod, build) {
+			if len(existingPod.OwnerReferences) > 0 {
+				// check to see if we have retried long enough for the old pod to get GC'ed
+				now := metav1.Now()
+				twoMinutesAgo := now.Add(-2 * time.Minute)
```

Comment:
nit - can simplify to 
```suggestion
				twoMinutesAgo := metav1.Now().Add(-2 * time.Minute)
```
no?